packages/sdk-codegen/src/python.gen.ts

```diff
@@ -183,17 +195,30 @@ ${this.hooks.join('\n')}
       : this.nullStr
   }
 
-  declareProperty(indent: string, property: IProperty) {
+  declareProperty(indent: string, property: IProperty, annotations = false) {
     const mappedType = this.typeMapModels(property.type)
     let propName = property.name
     if (this.pythonKeywords.has(propName)) {
       propName = propName + '_'
     }
     let propType = mappedType.name
     if (!property.required) {
-      propType = `Optional[${mappedType.name}] = ${this.nullStr}`
+      propType = `Optional[${mappedType.name}]`
+    }
+
+    let propDef
+    if (annotations) {
+      let annotation = propType
+      if (this.isBareForwardRef(property)) {
+        annotation = `ForwardRef(${propType})`
+      }
+      propDef = `${this.bumper(indent)}"${propName}": ${annotation}`
+    } else {
+      if (!property.required) {
+        propType += ` = ${this.nullStr}`
+      }
+      propDef = `${indent}${propName}: ${propType}`
```

Comment:
dual purposing `declareProperty` to also declare an entry in the `__annotations__` dict. If that feels too icky I could probably factor out the shared bit (getting the type and translating when needed) and have `declareProperty` and `declareAnnotation`(?) use it. I'll admit I was rushing through this part and reading it now it's kind of hard to tease apart what's what/when/where

Reply:
Did you need to change the signature of all the other `ICodeGen` implementations for this? Having it as a default `false` parameter seems ok to me. Make me think about "overloading" of annotations with the attributes for JSON property renames, but that's really an implementation detail someone could look at in this function.

Reply:
no, I was surprised that TS let me change this signature w/out complaining that the class no longer conformed to `ICodeGen` - I'm not deeply familiar with how that is supposed to work. Reading https://www.typescriptlang.org/docs/handbook/interfaces.html#implementing-an-interface didn't help... just a bit more confused

Reply:
LOL. Let's worry about it if it breaks :)